### PR TITLE
Improve selection of background video

### PR DESCRIPTION
**Pull request by westonruter**
* Limit video filetype to MP4
* Update “Edit Media“ to “Change Media” in button.
* Update warning to indicate the MB per second of the selected video. See https://github.com/ampproject/amp-wp/issues/2560#issuecomment-501117194.

### CODE REVIEW

File: assets/src/stories-editor/blocks/amp-story-page/edit.js [hunk 1]
```diff
@@ -302,6 +305,12 @@ class PageEdit extends Component {
 											VIDEO_ALLOWED_MEGABYTES_PER_SECOND
 										)
 									}
+									{
+										videoBytesPerSecond > 0 && ' ' + sprintf(
+											__( 'The selected video is %d MB per second.', 'amp' ),
```
Comment (swissspidy) [inline assets/src/stories-editor/blocks/amp-story-page/edit.js]: Missing translator comment

**Comment by westonruter** [comment]
# Testing instructions

Given an uploaded MP4 that is relatively large given its duration, such as:

image

Select the video as a page background and you should see the warning amended with text like “The selected video is 10 MB per second”:

image

If selecting an optimized video, such as:

image

Then no warning should appear:

image

File: assets/src/stories-editor/blocks/amp-story-page/edit.js (hunk 1 above)
Comment (westonruter) [reply assets/src/stories-editor/blocks/amp-story-page/edit.js]: Fixed in 1d20f42bc7bcb9d7e4adfca1c1ef2ffd60b4b86b

File: assets/src/stories-editor/helpers/index.js [hunk 2]
```diff
@@ -1425,6 +1425,19 @@ export const getCallToActionBlock = ( pageClientId ) => {
 	return innerBlocks.find( ( { name } ) => name === 'amp/amp-story-cta' );
 };
 
+/**
+ * Gets the number of megabytes per second for the video.
+ *
+ * @param {Object} media The media object of the video.
+ * @return {number} Number of megabytes per second, or -1 if media details unavailable.
```
Comment (swissspidy) [inline assets/src/stories-editor/helpers/index.js]: Why not just return `null` instead in that case? -1 is easy to miss and we don't use such a return type anywhere else.

Comment (westonruter) [reply assets/src/stories-editor/helpers/index.js]: Like so?

```diff
diff --git a/assets/src/stories-editor/blocks/amp-story-page/edit.js b/assets/src/stories-editor/blocks/amp-story-page/edit.js
index e5793fda..167f5a53 100644
--- a/assets/src/stories-editor/blocks/amp-story-page/edit.js
+++ b/assets/src/stories-editor/blocks/amp-story-page/edit.js
@@ -256,7 +256,7 @@ class PageEdit extends Component {
 
 		const colorSettings = this.getOverlayColorSettings();
 		const isExcessiveVideoSize = VIDEO_BACKGROUND_TYPE === mediaType && isVideoSizeExcessive( media );
-		const videoBytesPerSecond = VIDEO_BACKGROUND_TYPE === mediaType ? getVideoBytesPerSecond( media ) : -1;
+		const videoBytesPerSecond = VIDEO_BACKGROUND_TYPE === mediaType ? getVideoBytesPerSecond( media ) : null;
 
 		return (
 			<>
@@ -306,7 +306,7 @@ class PageEdit extends Component {
 										)
 									}
 									{
-										videoBytesPerSecond > 0 && ' ' + sprintf(
+										videoBytesPerSecond && ' ' + sprintf(
 											/* translators: %d: the number of actual megabytes per second */
 											__( 'The selected video is %d MB per second.', 'amp' ),
 											Math.round( videoBytesPerSecond / MEGABYTE_IN_BYTES )
diff --git a/assets/src/stories-editor/helpers/index.js b/assets/src/stories-editor/helpers/index.js
index 7d74cf03..669bc813 100644
--- a/assets/src/stories-editor/helpers/index.js
+++ b/assets/src/stories-editor/helpers/index.js
@@ -1429,11 +1429,11 @@ export const getCallToActionBlock = ( pageClientId ) => {
  * Gets the number of megabytes per second for the video.
  *
  * @param {Object} media The media object of the video.
- * @return {number} Number of megabytes per second, or -1 if media details unavailable.
+ * @return {number|null} Number of megabytes per second, or null if media details unavailable.
  */
 export const getVideoBytesPerSecond = ( media ) => {
 	if ( ! has( media, [ 'media_details', 'filesize' ] ) || ! has( media, [ 'media_details', 'length' ] ) ) {
-		return -1;
+		return null;
 	}
 	return media.media_details.filesize / media.media_details.length;
 };
```

**Comment by kienstra** [comment]
**Maybe possible to upload and select a .mov file**

Steps to reproduce
1. Click 'Background Media' > 'Select Media'
2. Upload a `.mov` file, [like this](https://drive.google.com/file/d/1Re9M8TJI_xvZ6zlWiMGe11B5Bbck9_Z9/view?usp=sharing)
3. Expected: it's not possible to select it, and there's no notice
4. Actual: it's possible so click 'Select' in the Media Library, and the notice appears (A video size of less than...)


notice-appears

**Comment by kienstra** [comment]
Ah, sorry. Not great timing to report that.

**Comment by swissspidy** [comment]
@kienstra So you're saying it (the mime type limitation) is not working?

**Comment by westonruter** [comment]
That seems like a bug in the media library. It should be respecting `allowedTypes` for both browsing and uploading.

**Comment by kienstra** [comment]
>@kienstra So you're saying it (the mime type limitation) is not working?

Yes, that seems to be the case. Though it's hard to say if it's an issue with this plugin.

I'm using Core version `5.2.2-alpha-45482` and Gutenberg version `5.9.0-rc.1`.

**Comment by westonruter** [comment]
It is a bug in core actually.

On the post editor I just tried adding an Image block. Then I went through the same flow, selecting a MOV file to upload and then immediately select:

image

I was then able to use the MOV as image, even though normally I would not be able to find MOV files by browsing the media modal.

**Comment by miina** [comment]
@csossi See https://github.com/ampproject/amp-wp/pull/2607#issuecomment-501465226 for testing instructions.

**Comment by csossi** [comment]
Verified in QA

**Pull request merged**